SimPEG/EM/Static/DC/IODC.py

```diff
@@ -0,0 +1,444 @@
+        required=True,
+        default=5
+    )
+
+    ncell_per_dipole = properties.Integer(
+        "The number of cells between dipole electrodes",
+        required=True,
+        default=5
+    )
+
+    # For synthetic surveys
+    x0 = None
+    lineLength = None
+    a = None
+    n_spacing = None
+
+    def from_ambn_locations_to_survey(
+        self, a_locations, b_locations, m_locations, n_locations,
+        survey_type=None, dobs=None,
+        data_type="volt", fname=None, dim=2
+    ):
+        """
+        read A, B, M, N electrode location and data (V or appResistivity)
+        """
+        geometric_factor = SimPEG.EM.Static.Utils.StaticUtils.geometric_factor
+
+        self.a_locations = a_locations.copy()
+        self.b_locations = b_locations.copy()
+        self.m_locations = m_locations.copy()
+        self.n_locations = n_locations.copy()
+        self.survey_type = survey_type
+        self.data_type = data_type
+        self.dim = dim
+
+        uniqSrc = Utils.uniqueRows(np.c_[self.a_locations, self.b_locations])
+        uniqElec = SimPEG.Utils.uniqueRows(
+            np.vstack((self.a_locations, self.b_locations, self.m_locations, self.n_locations))
+            )
+        self.electrode_locations = uniqElec[0]
+        nSrc = uniqSrc[0].shape[0]
+        ndata = self.a_locations.shape[0]
+
+        if self.survey_geometry == "SURFACE":
+            # 2D locations
+            srcLists = []
+            sortinds = []
+            for iSrc in range (nSrc):
+                inds = uniqSrc[2] == iSrc
+                sortinds.append(np.arange(ndata)[inds])
+
+                locsM = self.m_locations[inds, :]
+                locsN = self.n_locations[inds, :]
+
+                if (survey_type == 'dipole-dipole') or (survey_type == 'pole-dipole'):
+                    rx = Rx.Dipole_ky(locsM, locsN)
+                elif (survey_type == 'dipole-pole') or (survey_type == 'pole-pole'):
+                    rx = Rx.Pole_ky(locsM)
+
+                if dim == 2:
+                    locA = uniqSrc[0][iSrc, :2]
+                    locB = uniqSrc[0][iSrc, 2:]
+                elif dim == 3:
+                    locA = uniqSrc[0][iSrc, :3]
+                    locB = uniqSrc[0][iSrc, 3:]
+
+                if (survey_type == 'dipole-dipole') or (survey_type == 'dipole-pole'):
+                    src = Src.Dipole([rx], locA, locB)
+                elif (survey_type == 'pole-dipole') or (survey_type == 'pole-pole'):
+                    src = Src.Pole([rx], locA)
+
+                srcLists.append(src)
+
+            self.sortinds = np.hstack(sortinds)
+            survey = Survey_ky(srcLists)
+            self.a_locations = self.a_locations[self.sortinds, :]
+            self.b_locations = self.b_locations[self.sortinds, :]
+            self.m_locations = self.m_locations[self.sortinds, :]
+            self.n_locations = self.n_locations[self.sortinds, :]
+            self.G = geometric_factor(
```

Comment:
my suggestion here is to replace `G` with `geometric_factor` that is an `@property`. eg
```python
@property
def geomertic_factor(self):
    return geometric_factor(self.survey, survey_type=self.survey_type, space_type=self.space)
```

Reply:
I think we can have a function in the IO, but I still want to have G in the IO.